src/modules/TftpServer.rb

```diff
@@ -199,6 +199,7 @@ def Write
         # so we must kill it otherwise it will be using the old parameters
         Yast2::Systemd::Service.find!("tftp").stop
       else
+        socket.start if service.active?
```

Comment:
Doing that, the service starts correctly when the user choose the `Start` action from the service widget.

However, once the service is running, the `Restart` action ends with an stopped service, because of how the `SystemService` perform the restart action:

  - Stop both, socket and service
  - Start the needed one, the service **or** the socket, depending on the selected `start_mode`.

https://github.com/yast/yast-yast2/blob/62f62b02018caa3224f5f880aa62782e62554706/library/systemd/src/lib/yast2/system_service.rb#L509-L511

```ruby
    def perform_restart
      perform_stop && perform_start
    end
```

https://github.com/yast/yast-yast2/blob/62f62b02018caa3224f5f880aa62782e62554706/library/systemd/src/lib/yast2/system_service.rb#L478-L488

```ruby
    def perform_start
      result = true


      if socket && start_mode == :on_demand
        result &&= socket.start unless socket_active?
      else
        result &&= service.start unless service.active?
      end


      result
    end
```

So the service tries to start when the socket is stopped (the original issue).

Reply:
Another solution could be

```suggestion
        socket.start_mode = :on_demand if service.active?
```

which also "force" the "auto start mode". 

But, with this approach only the socket will be started, which I think it is enough... but I am not sure that this will be what the user expect (i.e., the user that report the bug).

Reply:
So, it seems to me that `SystemService` could have support for this kind of services that requires the socket running.

@imobachgs, @jreidinger, @joseivanlopez thoughts? advices?

Reply:
<sub>* I tried to take "animated gif" (aka videos), but I had problems with them.</sub>

Reply:
At first, let me say that I think it should be fixed in yast2 itself, but when I am reading bug report, for me it more looks like bug of tftp that it cannot run service without socket, which looks strange.

According to https://www.freedesktop.org/software/systemd/man/systemd.socket.html it is service responsibility to start socket if it is required for its run by adding dependency to its systemd file. So from my POV bug of tftp.

Reply:
> let me say that I think it should be fixed in yast2 itself,

I agree

> According to https://www.freedesktop.org/software/systemd/man/systemd.socket.html it is service responsibility to start socket if it is required for its run by adding dependency to its systemd file

Nice! Thanks for the link!

Reply:
> let me say that I think it should be fixed in yast2 itself,

I agree

> According to https://www.freedesktop.org/software/systemd/man/systemd.socket.html it is service responsibility to start socket if it is required for its run by adding dependency to its systemd file

Nice! Thanks for the link!